conda/faiss/build-lib.bat

```diff
@@ -14,6 +14,7 @@ cmake -B _build ^
       -A x64 ^
       -G "Visual Studio 16 2019" ^
       -DBUILD_SHARED_LIBS=ON ^
+      -DFAISS_ENABLE_C_API=ON ^
```

Comment:
The build failure on CI is related to windows not liking C_API since the C_API is not supported on that platform. Do you want to try not publishing C_API onto conda for windows mode and see if the CI passes?

Reply:
Thanks @mengdilin! Added a commit for this..